glusterd2/commands/volumes/brick-replace-txn.go

```diff
@@ -0,0 +1,98 @@
+package volumecommands
+
+import (
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+)
+
+func prepareBricks(c transaction.TxnCtx) error {
+	var req api.BrickReq
+	if err := c.Get("newBrick", &req); err != nil {
+		return err
+	}
+	err := PrepareBrick(req, c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func replaceVolinfo(c transaction.TxnCtx) error {
+	var newBrick api.BrickReq
+	if err := c.Get("newBrick", &newBrick); err != nil {
+		return err
+	}
+	var srcBrickInfo brick.Brickinfo
+	if err := c.Get("srcBrickInfo", &srcBrickInfo); err != nil {
+		return err
+	}
+	var subVolIndex int
+	if err := c.Get("subVolIndex", &subVolIndex); err != nil {
+		return err
+	}
+	var brickIndex int
+	if err := c.Get("brickIndex", &brickIndex); err != nil {
+		return err
+	}
+	var volInfo volume.Volinfo
+	if err := c.Get("volinfo", &volInfo); err != nil {
+		return err
+	}
+
+	// Replace brick details in original VolInfo
+	newBricks := []api.BrickReq{newBrick}
+	newBrickInfos, err := volume.NewBrickEntries(newBricks, srcBrickInfo.VolumeName, srcBrickInfo.VolfileID, srcBrickInfo.VolumeID, srcBrickInfo.PType)
+	if err != nil {
+		return err
+	}
+	newBrickInfo := newBrickInfos[0]
+
+	// Retaining the brick position same as old brick
+	volInfo.Subvols[subVolIndex].Bricks[brickIndex] = newBrickInfo
```

Comment:
please do length check before accessing the index  to avoid a panic

Reply:
The volInfo  already exists and I have not modified it, the index is retrieved from the same volInfo.